Comments/ClientApp/src/components/Answer/__tests__/Answer.test.js

```diff
@@ -1,112 +1,56 @@
-			const wrapper = mount(<Answer {...answerPropsWithAnswer} />);
-			expect(wrapper.find("textarea").length).toEqual(1);
-			expect(wrapper.find("textarea").props().defaultValue).toEqual(
-				"some answer text",
-			);
-		});
-
-		it("unsavedChanges function is fired correctly on text area change", () => {
-			const wrapper = mount(<Answer {...answerPropsWithAnswer} />);
-			expect(wrapper.state().unsavedChanges).toEqual(false);
-			const textArea = wrapper.find("textarea");
-			textArea.simulate("input", {
-				target: {
-					value: "an updated answer",
-				},
-			});
-			expect(wrapper.state().answer.answerText).toEqual("an updated answer");
-			expect(answerPropsWithAnswer.updateUnsavedIds).toHaveBeenCalledWith(
-				"22q",
-				true,
-			);
-		});
-
-		it("should update UnsavedChanges if lastupdateddate has changed", () => {
-			const wrapper = mount(<Answer {...answerPropsWithAnswer} />);
-			wrapper.setState({ unsavedChanges: true });
-			const updatedProps = {
-				answer: {
-					answerId: answerWithoutAnswer.answerId,
-					answerText: "an updated answer",
-					lastModifiedDate: new Date("02/04/2018").toISOString(),
-				},
-			};
-			wrapper.setProps(updatedProps);
-			expect(wrapper.state().unsavedChanges).toEqual(false);
-		});
-
-		it("should not update UnsavedChanges if lastupdateddate has not changed", () => {
-			const wrapper = mount(<Answer {...answerPropsWithAnswer} />);
-			wrapper.setState({ unsavedChanges: true });
-			wrapper.setProps(answerPropsWithAnswer);
-			expect(wrapper.state().unsavedChanges).toEqual(true);
-		});
+const answerPropsWithAnswer = {
+	answer: answerWithAnswer,
+	readOnly: false,
+	saveHandler: jest.fn(),
+	deleteHandler: jest.fn(),
+	unique: "string",
+	updateUnsavedIds: jest.fn(),
+	questionType: {
+		type: "Text",
+	},
+};
+
+const answerPropsWithoutAnswer = {
+	answer: answerWithoutAnswer,
+	readOnly: false,
+	saveHandler: jest.fn(),
+	deleteHandler: jest.fn(),
+	unique: "string",
+	updateUnsavedIds: jest.fn(),
+	questionType: {
+		type: "Text",
+	},
+};
+
+test("sets text area with comment text correctly", async () => {
+	render(<Answer {...answerPropsWithAnswer} />);
+	expect(screen.getByDisplayValue("some answer text")).toBeInTheDocument();
+});
 
-		it("updated comment text in state after new props received", () => {
-			const wrapper = mount(<Answer {...answerPropsWithAnswer} />);
-			const updatedProps = {
-				answer: {
-					answerId: answerWithoutAnswer.answerId,
-					answerText: "an updated answer",
-				},
-			};
-			wrapper.setProps(updatedProps);
-			expect(wrapper.state().answer.answerText).toEqual("an updated answer");
-		});
+test("unsavedChanges function is fired correctly on text area change", async () => {
+	render(<Answer {...answerPropsWithAnswer} />);
+	const textArea = screen.getByDisplayValue("some answer text");
+	const user = userEvent.setup();
+	textArea.focus();
+	await user.type(textArea, " that's been updated");
```

Comment:
space at start of typed text?

Reply:
Yeah, I wanted the test to replace the original value of 'some answer text', but it wouldn't clear this for some reason. The typing appends the text so the ultimate value would be:

`some answer text that's been updated`

Would it be better to move the space to the end of the existing value instead do you think?

Reply:
Might look a bit less odd in the test yes